components/AddPhotosGrid/AddPhotosGrid.js

```diff
@@ -44,9 +45,9 @@ class AddDogPhotosGrid extends React.Component {
         const amountOfCurrentPhotos = 6 - this.state.uploadedPhotos.length
         let correctDogImages = this.state.uploadedPhotos
         if (amountOfCurrentPhotos > 0) {
-            correctDogImages =  [...this.dogImages, ...this.createButtons(amountOfCurrentPhotos)] 
+            correctDogImages =  [...this.dogImages(), ...this.createButtons(amountOfCurrentPhotos)] 
```

Comment:
nice changing this over to a function so it will trigger render for an updated state!

Reply:
I got the idea from this genius software developer